### ISSUE TITLE
History is getting spammed when Color Appearance & Lighting module is enabled Build 5.10-190 on dev branch.

### ISSUE THREAD

**Issue by kaesa** [Beep6581/RawTherapee#7055]
**Short description**
When the Color Appearance & Lighting module is enabled and you modify any values in any modules, the history is getting spammed with CAL-VC-Tint and CAL-VC-Temperature values, which have not changed from the previous. Also, you need to double click snapshots to get them selected, as on the first click, the focus moves to the History.

**Steps to reproduce**
1. Enable Color Appearance & Lighting module from the Advanced tab.
2. Modify sliders anywhere or switch between snapshots.
3. History is filling up with lines of CAL-VC-Tint and CAL-VC-Temperature values, which have not changed from their previous values.

**Expected behavior**
History should stay cleaner to remain useful. One click on a snapshot should be enough to get it selected(for deleting).

**Additional information**
<details><summary>Details</summary>
<p>

Version: 5.10-190-gc72e67ae2
Branch: dev
Commit: c72e67ae2
Commit date: 2024-04-20
Compiler: cc 13.2.1
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.8
Lensfun: V0.3.4.0
Build type: Release
Build flags:  -std=c++11 -ffp-contract=off  -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags:   
OpenMP support: ON
MMAP support: ON
Build OS: Linux 5.15.154-1-MANJARO x86_64
Build date: Sat, 20 Apr 2024 15:59:44 +0000 UTC
Build epoch: 1713628784
Build UUID: a15bc86b-fbda-444d-bb2d-64476313b219

</p>
</details>

**Comment by Desmis** [comment]
@kaesa 

This is a (very) complex event management problem...This concerns the "automatic" management of the 2 sliders "Temp" and "Tint" in "Viewing conditions". And that makes several “issues” on this subject directly or indirectly, since many years.
Depending on the mode chosen, "classic", "symmetric", and depending on the "White point model" (WP)  chosen, the 2 cursors must or not be updated automatically. The addition of a 3rd "Mixt" mode - added to classic" and "symmetric" at the request of users has not reduced the complexity (quite the contrary).
So, in summary, I'm removing auto-update, it will be up to the user to make sure the values are consistent - which is the case in the overwhelming majority of cases. But I think, in a while, a request will appear on this subject (This must make 3 or 4 interventions on this link from memory).

I will open a PR soon

Jacques

**Comment by kaesa** [comment]
I would consider these bugs. I don't think the history is intended to get repeated lines of values that have not changed. The behavior of the snapshots module changes unexpectedly after the Color Appearance & Lighting module is enabled. The way to resolve or mitigate the problems is another thing.

**Comment by Desmis** [comment]
@kaesa 
Certainly...
But have you tried Pull-Request? History is only the tree that hides the forest
#7058 

Jacques

**Comment by kaesa** [comment]
I confirm the changes in your pull request corrects the behavior. I noticed before that there were some inconsistencies with the "Automatic" checkbox and the behavior kept changing, I mean changing in very wild ways, when enabling/disabling it. I didn't test it enough to make a clear report about it. Right now the checkbox doesn't do anything visible. I'm sorry if this is a work in progress and I'm giving excessive attention to a wrong thing.

**Comment by Desmis** [comment]
What I have been saying for several years, through various "issues", is that
* Ciecam is a complex concept...which is found practically nowhere. I think RT must probably be the only one to fully implement it
* Certain problems, like this one (this is the 4th issue on the subject) seem (at GUI level and with my weak skills) almost insoluble.

But I think we have to accept certain dysfunctions, if we want to have something other than the purely Cartesian aspects

jacques

**Comment by Desmis** [comment]
I just push a change, which can in some cases improve behavior

**Comment by kaesa** [comment]
I have tested the new commit and the behavior is correct regarding history and snapshots.

**Comment by Desmis** [comment]
@kaesa 
I think this change brings better overall behavior, based on the many possible possibilities.

If nothing stands in the way, I'll merge tomorrow
